setup.py

```diff
@@ -32,6 +32,7 @@
     "flake8",
     "black",
     "codecov",
+    "mypy",
```

Comment:
We should pin the mypy version for rechunker's CI:
https://mypy.readthedocs.io/en/stable/existing_code.html#continuous-integration